### ISSUE TITLE
AQL Error when using a custom ArangoSearch analyzer on a cluster deployment

### ISSUE THREAD

**Issue by jebbench** [arangodb/arangodb#10601]
## My Environment

* __ArangoDB Version__:        3.5.2
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Cluster / Oasis
* __Deployment Strategy__:   Kubernetes / Oasis
* __Configuration__:               3 node cluster with replication factor of 3
* __Infrastructure__:               GKE / Oasis
* __Operating System__:        Docker Container / oasis
* __Total RAM in your machine__:       Containers have 4GB
* __Disks in use__:        SSD / Oasis
* __Used Package__:              Docker - official Docker library / Oasis

## Component, Query & Data

__Affected feature__:
ArangoSearch / AQL query

__AQL query (if applicable)__:
```
FOR s IN `v-site`
  SEARCH ANALYZER(
  s.attributes.store_name.value == "my store name",
  "default::attribute"
  )
  RETURN s
```

__AQL explain (if applicable)__:
```
Query String (126 chars, cacheable: true):
 FOR s IN `v-site`
   SEARCH ANALYZER(
   s.attributes.store_name.value == "my store name",
   "default::attribute"
   )
   RETURN s

Execution plan:
 Id   NodeType            Site  Est.   Comment
  1   SingletonNode       DBS      1   * ROOT
  2   EnumerateViewNode   DBS      1     - FOR s IN v-site SEARCH ANALYZER((s.`attributes`.`store_name`.`value` == "my store name"), "default::attribute")   /* view query */
  6   RemoteNode          COOR     1       - REMOTE
  7   GatherNode          COOR     1       - GATHER 
  3   ReturnNode          COOR     1       - RETURN s

Indexes used:
 none

Optimization rules applied:
 Id   RuleName
  1   handle-arangosearch-views
  2   scatter-arangosearch-view-in-cluster
  3   remove-unnecessary-remote-scatter
```

__Dataset__:
Documents all match the format show below (note that this document has had most of the entries from the attributes array removed, an average document has about 100 entries in the array):
```json
{
  "jobId": "9a04216a-3290-4c03-8dbd-383993080485",
  "uniqueIdentifier": "1.0",
  "attributes": {
    "store_number": {
      "key": "store_number",
      "value": 1,
      "type": "number"
    },
	...
    "spinner": {
      "key": "spinner",
      "value": 2,
      "type": "number"
    }
  },
  "createdAt": "2019-11-29T10:41:46.777985Z",
  "updatedAt": "2019-11-29T10:41:46.778018Z"
}
```

__Size of your Dataset on disk__:
Not sure - tiny, ten thousand documents at less than 5KB each.

__Replication Factor & Number of Shards (Cluster only)__:
No shards, replication factor of 3 on the collection.

## Steps to reproduce

1. Run the query
2. Get the error:
```
Query: AQL: Error message received from cluster node 'PRMR-mjlklfpg': failed to build filter while querying arangosearch view, query '{"type":"n-ary or","typeID":63,"subNodes":[{"type":"n-ary and","typeID":62,"subNodes":[{"type":"function call","typeID":47,"name":"ANALYZER","subNodes":[{"type":"array","typeID":41,"subNodes":[{"type":"compare ==","typeID":25,"excludesNull":false,"subNodes":[{"type":"attribute access","typeID":35,"name":"value","subNodes":[{"type":"attribute access","typeID":35,"name":"store_name","subNodes":[{"type":"attribute access","typeID":35,"name":"attributes","subNodes":[{"type":"reference","typeID":45,"name":"s","id":0}]}]}]},{"type":"value","typeID":40,"value":"my store","vType":"string","vTypeID":4}]},{"type":"value","typeID":40,"value":"default::attribute","vType":"string","vTypeID":4}]}]}]}]}': 'ANALYZER' AQL function: Unable to lookup analyzer 'default::attribute' (while executing)
```
3. Edit the query (e.g. change `s.attributes.store_name.value` to `s.attributes.store_name.key`; doesn't seem to matter )
4. Run the query successfully a few times
5. Wait 30 seconds
6. Run the query and get the error again

__Problem__:

I have a query that has been working fine locally against a single instance docker deployment, when this query is run against a cluster deployment of Arango it errors.

Oddly if I edit the query and run it the query will return the expected results for about 30 seconds before it starts to fail again.

The analyzer is created as below (using the Java driver):
```kotlin
val attrAnalyzer = AnalyzerEntity()
attrAnalyzer.name = "attribute"
attrAnalyzer.type = AnalyzerType.norm
attrAnalyzer.properties = mapOf(Pair("locale", "en"), Pair("case", "lower"), Pair("accent", false))
db.createAnalyzer(attrAnalyzer)
```

__Expected result__:
The query should run reliably on a clustered deployment.

**Comment by jebbench** [comment]
The view definition:
```json
{
  "writebufferIdle": 64,
  "writebufferActive": 0,
  "type": "arangosearch",
  "primarySort": [
    {
      "field": "createdAt",
      "asc": false
    }
  ],
  "writebufferSizeMax": 33554432,
  "commitIntervalMsec": 1000,
  "consolidationPolicy": {
    "type": "tier",
    "segmentsBytesFloor": 2097152,
    "segmentsBytesMax": 5368709120,
    "segmentsMax": 10,
    "segmentsMin": 1,
    "minScore": 0
  },
  "globallyUniqueId": "c6011004/",
  "cleanupIntervalStep": 2,
  "id": "6011004",
  "links": {
    "site": {
      "analyzers": [
        "identity"
      ],
      "fields": {
        "uniqueIdentifier": {},
        "jobId": {},
        "attributes": {
          "analyzers": [
            "attribute"
          ],
          "includeAllFields": true,
          "storeValues": "id"
        },
        "createdAt": {}
      },
      "includeAllFields": false,
      "storeValues": "none",
      "trackListPositions": false
    }
  },
  "consolidationIntervalMsec": 10000
}
```

**Comment by gnusi** [comment]
Hi @jebbench, could you please post output of the following command:
```
db._useDatabase("_system");
db._databases();
```

**Comment by jebbench** [comment]
```
http+ssl://localhost:8529@_system> db._useDatabase("_system");
true

http+ssl://localhost:8529@_system> db._databases();
[
  "default",
  "_system"
]
```

I've tried upgrading to 3.5.3 and I still get the same error.

**Comment by gnusi** [comment]
@jebbench thanks, could you please execute another snippet:
```
var a = require("@arangodb/analyzers")
db._useDatabase("_system")
a.toArray();
db._useDatabase("default");
a.toArray();
```

**Comment by jebbench** [comment]
```
http+ssl://localhost:8529@_system> var a = require("@arangodb/analyzers")

http+ssl://localhost:8529@_system> db._useDatabase("_system")
true

http+ssl://localhost:8529@_system> a.toArray();
[
  [ArangoAnalyzer "text_sv" (type text)],
  [ArangoAnalyzer "text_ru" (type text)],
  [ArangoAnalyzer "text_nl" (type text)],
  [ArangoAnalyzer "text_no" (type text)],
  [ArangoAnalyzer "text_it" (type text)],
  [ArangoAnalyzer "text_de" (type text)],
  [ArangoAnalyzer "text_en" (type text)],
  [ArangoAnalyzer "identity" (type identity)],
  [ArangoAnalyzer "text_pt" (type text)],
  [ArangoAnalyzer "text_es" (type text)],
  [ArangoAnalyzer "text_zh" (type text)],
  [ArangoAnalyzer "text_fi" (type text)],
  [ArangoAnalyzer "text_fr" (type text)]
]

http+ssl://localhost:8529@_system> db._useDatabase("default");
true

http+ssl://localhost:8529@default> a.toArray();
[
  [ArangoAnalyzer "text_sv" (type text)],
  [ArangoAnalyzer "text_ru" (type text)],
  [ArangoAnalyzer "text_nl" (type text)],
  [ArangoAnalyzer "text_no" (type text)],
  [ArangoAnalyzer "text_it" (type text)],
  [ArangoAnalyzer "text_de" (type text)],
  [ArangoAnalyzer "text_en" (type text)],
  [ArangoAnalyzer "identity" (type identity)],
  [ArangoAnalyzer "text_pt" (type text)],
  [ArangoAnalyzer "text_es" (type text)],
  [ArangoAnalyzer "text_zh" (type text)],
  [ArangoAnalyzer "text_fi" (type text)],
  [ArangoAnalyzer "text_fr" (type text)],
  [ArangoAnalyzer "default::attribute" (type norm)]
]
```

**Comment by gnusi** [comment]
@jebbench In context of which database are you trying to execute a query that fails?

**Comment by jebbench** [comment]
Within the `default` database.

**Comment by gnusi** [comment]
@jebbench thank you for your help, I can confirm a bug with analyzers cache in cluster environment. The fix will be available in the upcoming major (3.6) and minor (3.5.4) releases.

**Comment by aveiros** [comment]
@gnusi does this affect Active Failover?

**Comment by gnusi** [comment]
@aveiros no, I'd say only cluster

**Comment by gnusi** [comment]
@aveiros on only on Oasis :-)

**Comment by gnusi** [comment]
@jebbench After investigation I can claim that issue doesn't exist in `3.6` (RC1 is available) and `devel`. For `3.5` fix will be available in `3.5.4`.

As the workaround for `3.5.3` I can suggest to create a global analyzer (in `_system` database) with the same name and properties as in `default` database.

I will notify you once fix will be merged into `3.5` branch.

**Comment by maxkernbach** [comment]
Hi @jebbench,

Thanks for reporting this issue, a fix is available in v3.5.4

**Comment by iMoses** [comment]
i'm having the same issue today on version 3.6.1

can it be related?
